python/pyspark/streaming/dstream.py

```diff
@@ -602,13 +602,17 @@ class TransformedDStream(DStream):
     Multiple continuous transformations of DStream can be combined into
     one transformation.
     """
-    def __init__(self, prev, func):
+    def __init__(self, prev, func, transformFunc=None):
```

Comment:
Why is this a parameter in the DStream constructor? Seems pretty weird to suddenly have one of the operations be randomly configurable through a field in the constructor. Why not override it in a subclass?